skimage/measure/tests/test_regionprops.py

```diff
@@ -476,18 +476,25 @@ def test_cache():
 
 
 def test_docstrings_and_props():
+    def foo():
+        """foo"""
+
+    has_docstrings = bool(foo.__doc__)
+
     region = regionprops(SAMPLE)[0]
 
     docs = _parse_docs()
     props = [m for m in dir(region) if not m.startswith('_')]
 
     nr_docs_parsed = len(docs)
     nr_props = len(props)
-    assert_equal(nr_docs_parsed, nr_props)
-
-    ds = docs['weighted_moments_normalized']
-    assert 'iteration' not in ds
-    assert len(ds.split('\n')) > 3
+    if has_docstrings:
+        assert_equal(nr_docs_parsed, nr_props)
+        ds = docs['weighted_moments_normalized']
+        assert 'iteration' not in ds
+        assert len(ds.split('\n')) > 3
+    else:
```

Comment:
Why not create bar with no doc string?

Reply:
don't get you, how would you detect that it has been dropped when in -OO then ?

Reply:
Basically -OO doesn’t assign that string to the property .doc. It is the same as simply omitting a docstring.

Reply:
> Basically -OO doesn’t assign that string to the property .doc. It is the same as simply omitting a docstring.

you got it :)